neurobooth_os/tasks/mbient_reset.py

```diff
@@ -65,41 +71,67 @@ def __init__(
             )
 
     def run(self, **kwargs):
-        # Send message to GUI terminal
-        print(f'Mbient Reset: {self.reset_key.upper()} to trigger reset, {self.continue_key.upper()} to continue.')
+        self.task_state: TaskState = TaskState.RESET_NO_SUCCESS
+        self.update_message()  # Present Intro Screen
 
-        # Present Intro Screen
-        self.update_message()
+        while self.task_state != TaskState.END_SCREEN:
+            if self.task_state == TaskState.RESET_NO_SUCCESS:
+                self.task_state = self.present_reset_no_success()
+            elif self.task_state == TaskState.RESET_POST_SUCCESS:
+                self.task_state = self.present_reset_post_success()
+
+        if self.show_end_screen:
+            self.present_end_screen()
+
+    def present_reset_no_success(self) -> TaskState:

```

Comment:
I think this is sufficiently complex to warrant method comments. Is there a diagram of the state transitions somewhere?

Reply:
Added comment at the top of the class with state transitions and rationale. All of this is to ensure that pressing ENTER will always perform a reasonable default behavior. (I.e., trigger another reset for the first time or if there was a failure, or simply advance to the end screen if there was a success.) Let me know if the new comments help to clear things up!